components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc

```diff
@@ -1,23 +1,30 @@
 [id="starknet_account_structure"]
 = Starknet account interface
 
-A Starknet account contract must include the following two functions:
+Starknet's account type is inspired by Ethereum's EIP-4337, where instead of EOAs, you use smart contract accounts with arbitrary verification logic.
 
-* `+__validate__+`
-* `+__execute__+`
+Using smart contracts provides complete flexibility within your account implementation.
 
-These functions serve distinct purposes to ensure that only the account owner can initiate transactions and that fees can be charged for the resources they use.
+While not mandatory at the protocol level, you can use a richer standard interface for accounts, defined in link:https://github.com/ericnordelo/SNIPs/blob/feat/standard-account/SNIPS/snip-6.md[Starknet Improvement Proposal #6 (SNIP-6)]. SNIP-6 was developed by community members at OpenZeppelin, in close collaboration with wallet teams and other Core Starknet developers.
 
-Starknet's account type is inspired by Ethereum's EIP-4337, where instead of an account being an EOA, it is now a smart contract account with arbitrary verification logic.
+[#account_entrypoints]
+== Account entrypoints
 
-Through the use of smart contracts, you are provided with complete flexibility within your account implementation.
+The following entrypoints are part of account contracts:
 
-While not mandatory at the protocol level, a richer standard interface for accounts was developed by the community. This standard was developed by OpenZeppelin, in a close collaboration with wallet teams and other Core Starknet developers. You can see the traits as defined in https://github.com/ericnordelo/SNIPs/blob/feat/standard-account/SNIPS/snip-6.md[Starknet Improvement Proposal #6 (SNIP-6)].
+[horizontal,labelwidth="35",role="stripes-odd"]
+`+__validate__+`:: (Required)
+`+__execute__+`:: (Required)
+`+__validate_declare__+`:: (Optional)
+`+__validate_deploy__+`:: (Optional)
 
-[id="replay_protection"]
+
+`+__validate__+` and `+__execute__+` ensure that only the account owner can initiate transactions, and that fees can be charged for the resources that the account uses.
```

Comment:
```suggestion
`+__validate__+` and `+__execute__+` in most account contracts typically ensure that only the account owner can initiate transactions, and that fees can be charged for the resources that the account uses.
```
At the protocol level, both functions can have arbitrary logic.

Reply:
@odednaor Fixed.